pkg/kubelet/prober/prober_manager.go

```diff
@@ -287,6 +287,12 @@ func (m *manager) isContainerStarted(pod *v1.Pod, containerStatus *v1.ContainerS
 }
 
 func (m *manager) UpdatePodStatus(pod *v1.Pod, podStatus *v1.PodStatus) {
+	containersWithReadinessProbe := sets.Set[string]{}
+	for _, container := range pod.Spec.Containers {
+		if container.ReadinessProbe != nil {
+			containersWithReadinessProbe.Insert(container.Name)
+		}
+	}
```

Comment:
This code is duplicated for the initContainers below. Would it make sense to wrap it into function?